### PR TITLE
Use arg_spec type for comparisons on default and choices

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
Differences between `default` and `choices` in `DOCUMENTATION` and `argument_spec` is a mixed bucket.

Some modules use strings for a default with a type of `int`, and other similar situations.  This works, but previously validate-modules was more strict there.  If `DOCUMENTATION` was an `int` and `argument_spec` was a `str`, and `type='int'` validate-modules would error.

This PR aims to use the type checkers from `AnsibleModule` when doing comparisons to convert data.

As a side effect we have 3 new errors where docs or args values cannot be converted to the specific type.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
test/sanity/validate-modules

##### ANSIBLE VERSION

```
devel
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by sivel** [comment]
@gundalow we might need to look into these in a little more depth to ensure it's doing the right thing. The inconsistencies in modules made this a bit more complex than I imagined.

**Comment by sivel** [comment]
I've spot checked a fair number of these, and am happy so far with the results.

File: test/sanity/validate-modules/main.py [hunk 1]
```diff
@@ -1097,13 +1132,25 @@ def _validate_argument_spec(self, docs):
                 )
 
             # TODO: needs to recursively traverse suboptions
-            doc_choices = docs.get('options', {}).get(arg, {}).get('choices', [])
-            if not compare_unordered_lists(data.get('choices', []), doc_choices):
+            try:
+                with CaptureStd():
+                    doc_choices = [_type_checker(c) for c in docs.get('options', {}).get(arg, {}).get('choices', [])]
```
Comment (mattclay) [inline test/sanity/validate-modules/main.py]: Would it be worthwhile to report on each choice separately?

Comment (sivel) [reply test/sanity/validate-modules/main.py]: Stop catching me trying to be lazy ;) 

It probably would be.

**Review by mattclay — APPROVED** [review]

File: test/sanity/validate-modules/main.py (hunk 1 above)
Comment (sivel) [reply test/sanity/validate-modules/main.py]: I've made the change to report on each choice separately

**Pull request merged**